models/coded.go

```diff
@@ -38,6 +38,14 @@ func (c *Coded) AddCodeIfPresent(codeElement xml.Node) {
 		codeSystem = CodeSystemFor(codeElement.Attribute("codeSystem").String())
 	}
 
+	//extract nullFlavor from attribute if it is NA
+	nullFlavorAttribute := codeElement.Attribute("nullFlavor")
+	nullFlavorValueSet := codeElement.Attributes()
```

Comment:
Can you just do `nullFlavorValueSet := codeElement.Attributes("valueSet")`, and then simplify the usage of it value below?